content/200-concepts/100-components/01-prisma-schema/09-indexes.mdx

```diff
@@ -0,0 +1,242 @@
+- **Existing hash indexes (PostgreSQL):** earlier versions of Prisma did not support hash indexes in the schema. These need to be added before migrating your database.
+
+In each of the cases above unwanted changes to your database can be prevented by properly specifying these properties in your data model where necessary. **The easiest way to do this is to use `prisma db pull` to retrieve any existing constraints.** Alternatively, you could also add these arguments manually. This should be done before using `prisma db push` or `prisma migrate dev`.
+
+## Full text indexes (MySQL and MongoDB)
+
+The `fullTextIndex` preview feature provides support for introspection and migration of full text indexes in MySQL and MongoDB in version 3.6.0 and later. This can be configured using the `@@fulltext` attribute. Existing full text indexes in the database are added to your Prisma schema after introspecting with `db pull`, and new full text indexes added in the Prisma schema are created in the database when using Prisma Migrate. This also prevents validation errors in some database schemas that were not working before.
+
+For now we do not enable the full text search commands in the Prisma Client. They will be implemented separately, and the progress can be followed in the [MongoDB](https://github.com/prisma/prisma/issues/9413) and [MySQL](https://github.com/prisma/prisma/issues/10415) issues.
+
+### Enabling the `fullTextIndex` preview feature
+
+To enable the `fullTextIndex` preview feature, add the `fullTextIndex` feature flag to the `generator` block of the `schema.prisma` file:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextIndex"]
+}
+```
+
+### Examples
+
+The following example demonstrates adding a `@@fulltext` index to the `title` and `content` fields of a `Post` model:
+
+```prisma file=schema.prisma
+model Post {
+  id      Int    @id
+  title   String @db.VarChar(255)
+  content String @db.Text
+
+  @@fulltext([title, content])
+}
+```
+
+On MongoDB, the `fullTextIndex` preview feature can be combined with the `extendedIndexes` preview feature to add ascending or descending columns to your full-text index:
+
+```prisma file=schema.prisma
+generator js {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextIndex", "extendedIndexes", "mongoDb"]
+}
+
+model Post {
+  id      String @id @map("_id") @db.ObjectId
+  title   String @db.VarChar(255)
+  content String @db.Text
+
+  @@fulltext([title(sort: Desc), content])
+}
```

Comment:
Seems to me that `@db.VarChar(255)` or `@db.Text` only make sense for MySQL but the example says we are using MongoDB here
```suggestion
generator js {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextIndex", "extendedIndexes", "mongoDb"]
}

datasource db {
  provider = "mongodb"
  url          = env("DATABASE_URL")
}

model Post {
  id      String @id @map("_id") @db.ObjectId
  title   String
  content String
  
  @@fulltext([title(sort: Desc), content])
}
```